### ISSUE TITLE
Segmentation fault  sudo /*/Sniffnet.app/Contents/MacOS/sniffnet

### ISSUE THREAD

**Issue by itfidele** [GyulyVGC/sniffnet#176]
**Describe the bug**
[1]    15572 segmentation fault  sudo /*/Sniffnet.app/Contents/MacOS/sniffnet

**To Reproduce**
Describe steps to reproduce the bug, e.g.:
1. Download to MacOs Setup V1.1.4
2. Install 
3. Try to open sniffnet
4. it asks u permission to allow .sh to access the shell
5. after it breaks to this segmentation fault

**Expected behavior**
I expected to start using the app after installation
**Screenshots**
<img width="726" alt="Screenshot 2023-05-14 at 00 15 51" src="https://github.com/GyulyVGC/sniffnet/assets/30210556/c3a4bedc-8bbc-4d28-a673-034674dabda5">

**Desktop (please complete the following information):**
 - OS: MacOS Montrey Version 12.6

**Comment by GyulyVGC** [comment]
Can you verify you have installed the app in the right location? 
It should be installed in your Applications folder, which should be under the root folder of your Mac.

**Comment by GyulyVGC** [comment]
Could you try launching it without sudo mode?

`/*/Sniffnet.app/Contents/MacOS/sniffnet`

Without admin privileges you will not actually be able to see any packets, but this is just to see if the app starts. In that case, it could be a problem related to your sudoers file.

**Comment by itfidele** [comment]
I did launch it in sudo mode, i saw by default it does that if you see there on the screenshot i had to enter password to get the following bug

**Comment by GyulyVGC** [comment]
Yep I know. I was just wondering if you could try launching _without_ sudo, to see if the problem persists. 

This is just to try understanding the cause of the issue.

**Comment by itfidele** [comment]
sure here is what it's throwing 
<img width="767" alt="Screenshot 2023-05-15 at 00 10 58" src="https://github.com/GyulyVGC/sniffnet/assets/30210556/eebd6667-99ef-47fb-8540-9e7e0e358d19">

**Comment by GyulyVGC** [comment]
I cannot reproduce your segmentation fault.
It's weird, because some users reported a segfault in version 1.1.3 (#109 - #137) but that problem has been solved in v1.1.4.

**Comment by itfidele** [comment]
which OS did you test it from?

**Comment by itfidele** [comment]
I did install the latest version, as it doesn't open terminal now it's closing on open #196

**Comment by GyulyVGC** [comment]
> which OS did you test it from? here i was using Macbook Pro, OS: MacOS Montrey Version 12.6

Macbook Air, Monterey 12.4

But the problem is probably related to your graphical adapter even if it's weird: I know other people besides me running it smoothly on macOS

**Comment by itfidele** [comment]
> > which OS did you test it from? here i was using Macbook Pro, OS: MacOS Montrey Version 12.6
> 
> Macbook Air, Monterey 12.4
> 
> But the problem is probably related to your graphical adapter even if it's weird: I know other people besides me running it smoothly on macOS

Yes it might be it 

Macbook Pro, Monterey 12.6 
Graphics : Intel Iris Pro 1536 MB

**Comment by GyulyVGC** [comment]
Mine is `Intel HD Graphics 6000 1536 MB`
As a possible solution I can try building for you a version of the software using a renderer different from the default one.

**Comment by GyulyVGC** [comment]
Here you are @itfidele 

[sniffnet.dmg.zip](https://github.com/GyulyVGC/sniffnet/files/11515478/sniffnet.dmg.zip)

This is the same version (1.2.0) but built using the `glow` renderer instead of the `wgpu` renderer.
These are the two renderers currently used by [Iced](https://github.com/iced-rs/iced); `wgpu` is suggested to be set as the default one since it generally has more compatibility, but some hardware works only with `glow`.

I hope this is your case as well.

**Comment by itfidele** [comment]
> Here you are @itfidele
> 
> [sniffnet.dmg.zip](https://github.com/GyulyVGC/sniffnet/files/11515478/sniffnet.dmg.zip)
> 
> This is the same version (1.2.0) but built using the `glow` renderer instead of the `wgpu` renderer. These are the two renderers currently used by [Iced](https://github.com/iced-rs/iced); `wgpu` is suggested to be set as the default one since it generally has more compatibility, but some hardware works only with `glow`.
> 
> I hope this is your case as well.

@GyulyVGC  This one worked perfectly so this was why it kept breaking?,

any suggestion on how to change this during development as i tried to build and learn this code from source code and was facing the same problem

**Comment by itfidele** [comment]
if you could add this one on release page it would be great also, thanks a lot

**Comment by GyulyVGC** [comment]
> > Here you are @itfidele
> > [sniffnet.dmg.zip](https://github.com/GyulyVGC/sniffnet/files/11515478/sniffnet.dmg.zip)
> > This is the same version (1.2.0) but built using the `glow` renderer instead of the `wgpu` renderer. These are the two renderers currently used by [Iced](https://github.com/iced-rs/iced); `wgpu` is suggested to be set as the default one since it generally has more compatibility, but some hardware works only with `glow`.
> > I hope this is your case as well.
> 
> @GyulyVGC This one worked perfectly so this was why it kept breaking?,
> 
> any suggestion on how to change this during development as i tried to build and learn this code from source code and was facing the same problem

Yep, it was breaking for graphical adapter incompatibility evidently.

For development, just switch to the `glow-renderer` branch.

I'm so happy we solved the problem 💖

**Comment by itfidele** [comment]
@GyulyVGC  i think this issue is still there, i downloaded the latest and it was opening and closed by itself, the terminal is not appearing tho, i thought it might be the same as this issue they seem similar - Macbook Pro Intel

**Comment by GyulyVGC** [comment]
That's because in the release page it's only published the app built with the default renderer.

Tomorrow I'll provide you the version of the app that works for you 👍

(It's not a problem but a question of different graphical renderers, each one with its intrinsic compatibilities)

**Comment by GyulyVGC** [comment]
There you go @itfidele 

[Sniffnet_macOS_Intel.dmg.zip](https://github.com/GyulyVGC/sniffnet/files/11778555/Sniffnet_macOS_Intel.dmg.zip)
